ngx_http_auth_ldap_module.c

```diff
@@ -1334,18 +1345,31 @@ ngx_http_auth_ldap_ssl_handshake_handler(ngx_connection_t *conn, ngx_flag_t vali
           long chain_verified = SSL_get_verify_result(conn->ssl->connection);
 
           int addr_verified;
-          char *hostname = c->server->ludpp->lud_host;
-          addr_verified = X509_check_host(cert, hostname, 0, 0, 0);
-
-          if (!addr_verified) { // domain not in cert? try IP
-            size_t len; // get IP length
-            if (conn->sockaddr->sa_family == 4) len = 4;
-            else if (conn->sockaddr->sa_family == 6) len = 16;
-            else { // very unlikely indeed
-              ngx_http_auth_ldap_close_connection(c);
-              return;
+          if (c->server->ssl_check_cert == SSL_CERT_VERIFY_CHAIN) {
```

Comment:
handle `chain` (`SSL_CERT_VERIFY_CHAIN`) case: we've already called `SSL_get_verify_result` and got `chain_verified` assigned, which will be checked below. Set `addr_verified` to `1` (success) so it will get ignored in the checks later. Don't call `X509_check_host` or `X509_check_ip`.